apps/web/pages/api/book/request-reschedule.ts

```diff
@@ -104,9 +106,18 @@ const handler = async (
       },
     });
 
-    if (bookingToReschedule.userId !== userOwner.id) throw new Error("UNAUTHORIZED");
+    const bookingBelongsToTeam = !!bookingToReschedule?.eventType?.teamId;
+    if (bookingBelongsToTeam && bookingToReschedule?.eventType?.teamId) {
```

Comment:
Nit: instead of putting so may ?s, we can check for `bookingToReschedule?.eventType` once in the beginning and throw error maybe

Reply:
Agreed with @hariombalhara this also aligns with our [Prefer early returns](https://app.gitbook.com/o/6snd8PyPYMhg0wUw6CeQ/s/VXRprBTuMlihk37NQgUU/engineering/best-practices/prefer-early-returns) style